### PR TITLE
transform logging.enabled to logging.status

### PR DESCRIPTION
**Pull request by jafowler**
the terraform provider only looks for `logging.status` and thus if `logging.enabled` is the only thing that exists, we encounter a rulesets error when attempting to run terraform apply. This PR will look at the `enabled` field and translate it to the corresponding `status` field. 

Schema: https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/internal/provider/schema_cloudflare_ruleset.go#L964-L970

resource_cloudflare_ruleset: https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/internal/provider/resource_cloudflare_ruleset.go#L1284-L1296

### CODE REVIEW

**Comment by jafowler** [comment]
I was working on fixing the rest of the linting errors, and when i went and added error checks around the json.Marshal the tests started failing. 

The test that failed seemed to be mostly random, but it was always an index out of range error. I preserved the branch under https://github.com/cloudflare/cf-terraforming/tree/figure-out-why-tests-are-failing so we can continue to debug it without impacting this PR. @jacobbednarz do you have any thoughts on why adding error checks to `json.Marshal` started making things error out?

File: testdata/terraform/cloudflare_ruleset_zone_http_request_firewall_custom/test.tf [hunk 1]
```diff
@@ -15,6 +15,7 @@ resource "cloudflare_ruleset" "terraform_managed_resource" {
     }
     logging {
       enabled = true
```
Comment (jacobbednarz) [inline testdata/terraform/cloudflare_ruleset_zone_http_request_firewall_custom/test.tf:17]: `enabled = (bool)` shouldn't be present as they were the old approach before `status = "..."`; you'll want to unset them in the generator.

File: internal/app/cf-terraforming/cmd/generate.go [hunk 2]
```diff
@@ -769,6 +769,25 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
 									}
 								}
+								// do we have a logging parameter in our rule?
+								if rules.([]interface{})[ruleCounter].(map[string]interface{})["logging"] != nil {
+									// make sure it has data
+									if rules.([]interface{})[ruleCounter].(map[string]interface{})["logging"].(map[string]interface{})["enabled"] != nil {
+										/*
+											enabled = true -> status = 'enabled'
+											enabled = flase -> status = 'disabled'
+											Also lets only add 'status' if it doesn't already exist
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:779]: if it doesn't exist, it shouldn't be here as that is the default and we don't want to send it.

Comment (jafowler) [reply internal/app/cf-terraforming/cmd/generate.go:779]: in some cases, `status` won't exist when `enabled` is set. This is the case for the latest clone we tried to do. What i was trying to say in my comment, and maybe i need to make this more clear, is only set status if it doesn't exist and enabled is set.

**Review by manatarms — APPROVED** [review]

**Review by jacobbednarz — APPROVED** [review]

**Pull request merged**